recipe_scrapers/weightwatchers.py

```diff
@@ -0,0 +1,152 @@
+        return "www.weightwatchers.com"
+
+    def author(self):
+        return "WeightWatchers"
+
+    def title(self):
+        return self.soup.find("h1").get_text().strip()
+
+    def category(self):
+        return "WeightWatchers"
+
+    # cooking times, yield, difficulty are in a common div in public and non-public recipes
+    # but class of that block and sub elements are different
+    # so finding the block and extracting a value will be overridden in class for public recipes,
+    # but picking the data item based on order is don in this base class (total_time(), cook_time() and so on)
+    def _findDataContainer(self):
+        return self.soup.find("div", {"class": "styles_container__3N3E8"})
+
+    def _extractItemField(self, item):
+        return item.contents[1]
+
+    def total_time(self):
+        return get_minutes(
+            self._extractItemField(self._findDataContainer().contents[0])
+        )
+
+    def prep_time(self):
+        return get_minutes(
+            self._extractItemField(self._findDataContainer().contents[1])
+        )
+
+    def cook_time(self):
+        return get_minutes(
+            self._extractItemField(self._findDataContainer().contents[2])
+        )
+
+    def yields(self):
+        return get_yields(self._extractItemField(self._findDataContainer().contents[3]))
+
+    def difficulty(self):
+        return self._extractItemField(self._findDataContainer().contents[4]).get_text()
+
+    #   Alternative way to extract data based on description instead of position
+    #    def total_time(self):
+    #        return get_minutes(
+    #            self.__findDataContainer()
+    #            .find("div", string=re.compile(r"minutes Total Time"))
+    #            .previous_sibling
+    #        )
+
+    def image(self):
+        backgroundImgStyle = self.soup.find("div", {"class": "styles_image__2dnNm"})[
+            "style"
+        ]
+
+        if backgroundImgStyle:
+            return (
+                re.compile(r'url\("(?P<imgurl>\S*)"\);')
+                .search(backgroundImgStyle)
+                .groupdict()
+                .get("imgurl")
+            )
+
+        return None
+
+    def _findIngridientTags(self):
+        return self.soup.find(
+            "h3", {"id": "food-detail-recipe-ingredients-header"}
+        ).parent.find_all("div", {"class": "styles_name__1OYVU"})
+
+    def _extractIngridientName(self, ingridient):
+        return normalize_string(
+            ingridient.find("div", {"class": "styles_ingredientName__1Vffd"})
+            .find("div")
+            .get_text()
+        )
+
+    def _extractPortionParts(self, ingridient):
+        tags = ingridient.find("div", {"class": "styles_portion__2NQyq"}).find_all(
+            "span"
+        )
+
+        return (
+            normalize_string(tags[0].get_text()),
+            normalize_string(tags[1].get_text()),
+            normalize_string(tags[2].get_text().replace(", ", "")) if tags[2] else None,
```

Comment:
Is it possible that there isn't a `2` index in the `tags` array? This code would work correctly if the value exists, but is `Falsey`.

Reply:
All the recipies I checked had three span tags even when there was no comment. Then the 3rd one (tags[2]) was empty

Reply:
If you believe, by your observation of multiple recipes, that there will always be 3 span tags then I'm okay with making this assumption in the code. If you'd like to handle the potential index error you're welcome to choose the solution of your liking. Catching an exception and throwing [another one](https://github.com/hhursev/recipe-scrapers/blob/aacd321f96235e44e438202c76f7eceddf4aa193/recipe_scrapers/_exceptions.py#L2) might be the simplest in that you can continue to make your assumption, but the code will gracefully fail.

Reply:
Added some exception handling